src/util/address.rs

```diff
@@ -511,13 +513,34 @@ impl Address {
         }
     }
 
-    /// Create a pay to taproot address
-    pub fn p2tr(taptweaked_key: schnorrsig::PublicKey, network: Network) -> Address {
+    /// Create a pay to taproot address from untweaked key
+    pub fn p2tr<C: Verification>(
+        secp: Secp256k1<C>, 
+        internal_key: UntweakedPublicKey, 
+        merkle_root: Option<TapBranchHash>, 
+        network: Network
+    ) -> Address {
         Address {
             network: network,
             payload: Payload::WitnessProgram {
                 version: WitnessVersion::V1,
-                program: taptweaked_key.serialize().to_vec()
+                program: internal_key.tap_tweak(secp, merkle_root).into_inner().serialize().to_vec()
```

Comment:
nit: 
I think you should be able to call serialize without calling `into_inner()`